pkg/crc/api/api_http_test.go

```diff
@@ -0,0 +1,267 @@
+package api
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"strings"
+	"testing"
+
+	"github.com/code-ready/crc/pkg/crc/machine/fakemachine"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/stretchr/testify/require"
+)
+
+type mockServer struct {
+	*server
+}
+
+func newMockServer() *mockServer {
+	fakeMachine := fakemachine.NewClient()
+	config := setupNewInMemoryConfig()
+
+	handler := NewHandler(config, fakeMachine, &mockLogger{}, &mockTelemetry{})
+
+	return &mockServer{
+		server: newServerWithRoutes(handler),
+	}
+}
+
+func sendRequest(handler http.Handler, request *request) *http.Response {
+	url := fmt.Sprintf("/%s", request.resource)
+	var data io.Reader
+	if request.data != "" {
+		data = strings.NewReader(request.data)
+	} else {
+		data = nil
+	}
+
+	req := httptest.NewRequest(request.httpMethod, url, data)
+	req.Header.Set("Content-Type", "application/json")
+	{
+		requestDump, _ := httputil.DumpRequest(req, true)
+		fmt.Println(string(requestDump))
+	}
```

Comment:
@cfergeau why we need this in a separate block?

Reply:
Not sure, iirc I wanted to isolate visually this code during development in case I want to move it someplace else or something like this ^^ Definitely not needed, I can remove the extra `{}`

Reply:
@cfergeau I figrued, it just break the flow during review (I thought you only need for debug purpose and not need when put the PR). when I run the test cases looks like it is required because the level of info it add to test.

Reply:
It may fit better in the main api/ code rather than in tests (enabled by CRC_DEBUG=http-api or such), but I forgot to look into that :)